Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs

```diff
@@ -1,21 +1,21 @@
 
 namespace StreetCode.DAL.Repositories.Interfaces.Base;
 
-public interface IRepositoryBase
+public interface IRepositoryBase<T>
 {
 
-    public void Create();
+     void Create();
 
-    public void CreateAsync();
+     Task CreateAsync();
 
-    public void GetById();
+     void GetById();
 
-    public void GetAll();
+     IQueryable<T> GetAll();
 
-    public void GetAllAsync();
+     Task<IEnumerable<T>> GetAllAsync();
```

Comment:
Also, it wouldn't hurt to add optional Expressions/Predicates, bound to default/null by default.